packages/sources/oanda/src/endpoint/price.ts

```diff
@@ -0,0 +1,153 @@
+
+const logger = makeLogger('OandaPrice')
+
+let instrumentMap: InstrumentMap
+
+// Get mapping of all available instruments keyed by base and quote assets
+const setInstrumentMap = async (context: EndpointContext<ModifiedSseGenerics>) => {
+  logger.info({ msg: 'Setting instrument map' })
+  const { data, status } = await axios.get<InstrumentList>(
+    `${context.adapterConfig.INSTRUMENTS_API_ENDPOINT}/accounts/${context.adapterConfig.API_ACCOUNT_ID}/instruments`,
+    {
+      headers: {
+        contentType: 'application/json',
+        Authorization: `Bearer ${context.adapterConfig.SSE_API_KEY}`,
+      },
+    },
+  )
+
+  if (!data || status !== 200) {
+    throw new AdapterError({ message: 'Could not fetch asset list', providerStatusCode: status })
+  }
+
+  instrumentMap = data.instruments.reduce((instrumentMap: InstrumentMap, item) => {
+    const [base, quote] = item.name.split('_')
+    instrumentMap[base] = instrumentMap[base] ?? {}
+    instrumentMap[base][quote] = item.name
+    return instrumentMap
+  }, {})
+}
+
+// See https://developer.oanda.com/rest-live-v20/pricing-ep/
+const sseTransport = new ModifiedSseTransport<ModifiedSseGenerics>({
+  prepareSSEConnectionConfig: (subsList, context) => {
+    const url = `${context.adapterConfig.SSE_API_ENDPOINT}/accounts/${
+      context.adapterConfig.API_ACCOUNT_ID
+    }/pricing/stream?instruments=${subsList.join('%2C')}`
+    return {
+      url,
+      config: {
+        responseType: 'stream',
+        headers: {
+          Authorization: `Bearer ${context.adapterConfig.SSE_API_KEY}`,
+        },
+      },
+    }
+  },
+  parseSubscriptionList: async (subscriptions, context) => {
+    if (!instrumentMap) await setInstrumentMap(context)
+
+    return subscriptions.map(({ base, quote }) => instrumentMap?.[base]?.[quote] ?? undefined)
+  },
+  eventListeners: [
+    {
+      type: 'PRICE',
+      parseResponse: (
+        event: MessageEvent<ModifiedSseGenerics['Provider']['ResponseBody']>['data'],
+      ) => {
+        logger.info({ event })
+        const { bids, asks, instrument, time } = event
+        const liquidBid = new Decimal(bids[bids.length - 1].price)
+        const liquidAsk = new Decimal(asks[asks.length - 1].price)
+        const result = liquidBid.add(liquidAsk).div(2).toNumber()
+        const [base, quote] = instrument.split('_')
+        return [
+          {
+            params: { base, quote },
+            response: {
+              data: { result },
+              result,
+              timestamps: {
+                providerIndicatedTimeUnixMs: new Date(time).getTime(),
+              },
+            },
+          },
+        ]
+      },
+    },
+  ],
+})
+
+// See https://developer.oanda.com/exchange-rates-api/#get-/v2/rates/spot.-ext-
+const restTransport = new HttpTransport<HttpGenerics>({
+  prepareRequests: (params: PriceEndpointParams[], config) =>
+    params.map((p) => {
+      const { base, quote } = p
+
+      return {
+        params: [p],
```

Comment:
Shouldn't this just be `p` since the request is for a single set of params?
```suggestion
        params: p,
```

Reply:
The `ProviderRequestConfig` type requires this to be an array: https://github.com/smartcontractkit/ea-framework-js/blob/a3e9ba5a12d96a1b954c2d1bb990c9e19cd73173/src/transports/http.ts#L43

Reply:
Oh shoot. I was mixing up `prepareRequest` and `parseResponse`. I see you have it correct below.